src/main/java/com/ably/kafka/connect/ChannelSinkTask.java

```diff
@@ -42,55 +49,35 @@ AblyClient getAblyClient() {
     @Override
     public void start(Map<String, String> settings) {
         logger.info("Starting Ably channel Sink task");
-        try {
-            ablyClient = ablyClientFactory.create(settings);
-        } catch (ChannelSinkConnectorConfig.ConfigException e) {
-            logger.error("Failed to create Ably client", e);
-        }
-        ablyClient.connect(isSuspended -> {
-            synchronized(ChannelSinkTask.this){
-                suspended.set(isSuspended);
-                if (!isSuspended){
-                    processSuspendQueue();
-                }
-            }
-        });
-    }
 
-    @Override
-    public void put(Collection<SinkRecord> records) {
-        if (ablyClient == null) {
-            throw new ConnectException("Ably client is unitialized");
+        try {
+            this.ablyClient = (DefaultAblyBatchClient) this.ablyClientFactory.create(settings);
+            this.ablyClient.connect();
+        } catch (ChannelSinkConnectorConfig.ConfigException | AblyException e) {
+            throw new RuntimeException(e);
         }
 
-        for (final SinkRecord record : records) {
-            publishSingleRecord(record);
+        // start the Batch processing thread.
+        this.sinkRecords = new ConcurrentLinkedQueue<>();
+        this.batchProcessingThread = new BatchProcessingThread(this.sinkRecords, this.ablyClient);
+        this.executor = new BatchProcessingExecutor(Integer.parseInt(settings
+                .getOrDefault(ChannelSinkConnectorConfig.BATCH_EXECUTION_THREAD_POOL_SIZE,
+                        ChannelSinkConnectorConfig.BATCH_EXECUTION_THREAD_POOL_SIZE_DEFAULT)));
+
+        for(int i = 0; i < this.executor.getCorePoolSize(); i++) {
+            this.executor.scheduleAtFixedRate(this.batchProcessingThread, 0,
```

Comment:
I am not all that familiar with the Java `ScheduledThreadPoolExecutor` (which BatchProcessingExecutor inherits from), but AFAICT, the default `BATCH_EXECUTION_FLUSH_TIME` is 5s. Can you help me understand how that time period is used and what impact that has? Does that mean we only publish in 5s intervals per thread?

Reply:
Yes the thread will be called every 5 seconds.

Reply:
Thanks for confirming, but my question was not whether it is called every 5 seconds, but what your thinking is behind that and what impact that has on this connector? Does that mean we can expect delays of up to 5s for publishing for example. Please explain why this exists and what behaviour is expected as a result from this connector.

Reply:
Its the logic of having the thread wake up and process messages in the queue. The idea is to have the thread sleep and yield it to the CPU when it has nothing to do. Current implementation is its configurable by the user and its defined in milliseconds, we could set to a much lower value like 100 ms.